### PR TITLE
feat: add qr code to register nano contract

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- It should create a navigation stack to contain:
  - A screen to read QR Code
  - A screen to inform the nano contract to register, which should be fulfilled by QR Code value read
  - A screen to ask camera permission
- It should create a screen to read QR Code of Nano Contract ID
- It should adjust the NanoContractRegisterScreen to make it accept a Nano Contract ID on its route param

Registering Nano Contract by QR Code reading:

https://github.com/user-attachments/assets/fe5ba910-3416-493c-822d-dab7547b2336

Closes: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/519

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/screens/NanoContractRegisterQrCodeScreen.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          justifyContent: 'center',
+          alignItems: 'center',
+          margin: 16,
+          alignSelf: 'stretch',
+        }}
+        >
+          <QRCodeReader
+            navigation={navigation}
+            onSuccess={onSuccess}
+            bottomText={t`Scan the token QR code`}
```
Comment (andreabadesso) [inline src/screens/NanoContractRegisterQrCodeScreen.js:63]: Missing translation for this file

File: src/screens/NanoContractRegisterQrCodeScreen.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    >
+      <View style={{
+        flex: 1,
+        justifyContent: 'center',
+        alignItems: 'center',
+        alignSelf: 'stretch',
+      }}
+      >
+        <HathorHeader
+          title={t`Register Nano Contract`.toUpperCase()}
```
Comment (andreabadesso) [inline src/screens/NanoContractRegisterQrCodeScreen.js:48]: Can you change this to `Nano Contract Registration`? Or change the one in the previous screen to `Register Nano Contract` so they're consistent when navigating?

**Review by andreabadesso — CHANGES_REQUESTED** [review]
Missing translations

File: locale/texts.pot [hunk 3]
```diff
@@ -1380,15 +1383,19 @@ msgstr ""
...
 
-#: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:93
+#: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:98
 msgid "Arguments"
 msgstr ""
 
+#: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:103
+msgid "Loading arguments."
+msgstr ""
+
```
Comment (pedroferreira1) [inline locale/texts.pot:1398]: Is this from this PR?

File: locale/texts.pot [hunk 4]
```diff
@@ -1510,19 +1517,33 @@ msgstr ""
...
+#: src/components/NanoContract/NanoContractTransactionActionListItem.js:108
+#, javascript-format
+msgid "Deposit ${ tokenSymbol }"
+msgstr ""
+
+#: src/components/NanoContract/NanoContractTransactionActionListItem.js:109
+msgid "Withdrawal ${ tokenSymbol }"
+msgstr ""
+
+#: src/components/NanoContract/NanoContractTransactionHeader.js:93
```
Comment (pedroferreira1) [inline locale/texts.pot:1537]: Why do you have this change in this PR?

File: src/App.js [hunk 5]
```diff
@@ -400,6 +467,8 @@ const AppStack = () => {
         />
         <Stack.Screen name='PaymentRequestDetail' component={PaymentRequestDetail} />
         <Stack.Screen name='RegisterToken' component={RegisterTokenStack} />
+        <Stack.Screen name='RegisterNanoContract' component={RegisterNanoContractStack} />
+        <Stack.Screen name='Register' component={RegisterTokenStack} />
```
Comment (pedroferreira1) [inline src/App.js:471]: Why do you need this? The token stack is already above

File: src/screens/NanoContractRegisterQrCodeScreen.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+          justifyContent: 'center',
+          alignItems: 'center',
+          margin: 16,
+          alignSelf: 'stretch',
+        }}
+        >
+          <QRCodeReader
+            navigation={navigation}
+            onSuccess={onSuccess}
+            bottomText={t`Scan the token QR code`}
```
Comment (pedroferreira1) [inline src/screens/NanoContractRegisterQrCodeScreen.js:63]:
```suggestion
            bottomText={t`Scan the nano contract ID QR code`}
```

File: src/App.js (hunk 5 above)
Comment (alexruzenhack) [reply src/App.js:471]: I must have forgotten it there.

File: locale/texts.pot (hunk 3 above)
Comment (alexruzenhack) [reply locale/texts.pot]: Probably it is because of i18n update.

File: locale/texts.pot (hunk 4 above)
Comment (alexruzenhack) [reply locale/texts.pot]: Probably it is because of i18n update.

File: src/screens/NanoContractRegisterQrCodeScreen.js (hunk 6 above)
Comment (alexruzenhack) [reply src/screens/NanoContractRegisterQrCodeScreen.js:63]: Done.

File: locale/pt-br/texts.po [hunk 7]
```diff
@@ -503,6 +503,22 @@ msgstr "Saldo Disponível"
...
+msgstr "Digitar"
+
+#: src/screens/NanoContract/NanoContractRegisterScreen.js:251
+#: src/screens/NanoContractRegisterQrCodeScreen.js:48
+msgid "Nano Contract Registration"
+msgstr "Registro do Nano Contract"
+
+#: src/screens/NanoContractRegisterQrCodeScreen.js:63
+msgid "Scan the nano contract ID QR code"
+msgstr "Leia o QR code Nano Contract ID"
```
Comment (pedroferreira1) [inline locale/pt-br/texts.po:520]:
```suggestion
msgstr "Leia o QR code do Nano Contract ID"
```

**Review by pedroferreira1 — APPROVED** [review]

File: locale/ru-ru/texts.po [hunk 8]
```diff
@@ -492,6 +492,22 @@ msgstr "Доступный Баланс"
 msgid "Locked"
 msgstr "Заблокированный"
 
+#.  translator: Used when the QR Code Scanner is opened, and user will manually
+#.  enter the information.
+#: src/screens/NanoContractRegisterQrCodeScreen.js:26
+#: src/screens/RegisterToken.js:27 src/screens/SendScanQRCode.js:87
+msgid "Manual info"
+msgstr "Ручной ввод"
```
Comment (andreabadesso) [inline locale/ru-ru/texts.po:500]: Was this an automatic translation? Maybe remove?

**Review by andreabadesso — APPROVED** [review]

File: locale/ru-ru/texts.po (hunk 8 above)
Comment (alexruzenhack) [reply locale/ru-ru/texts.po:500]: No, this was already there. I believe it was only moved.

**Pull request merged**